src/plugins/vis_default_editor/public/schemas.ts

```diff
@@ -17,35 +17,12 @@
  * under the License.
  */
 
-import { ReactNode } from 'react';
 import _, { defaults } from 'lodash';
 
 import { Optional } from '@kbn/utility-types';
 
-import { AggGroupNames, AggParam, AggGroupName } from '../../data/public';
-
-export interface ISchemas {
-  [AggGroupNames.Buckets]: Schema[];
-  [AggGroupNames.Metrics]: Schema[];
-  all: Schema[];
-}
-
-export interface Schema {
-  aggFilter: string[];
-  editor: boolean | string;
-  group: AggGroupName;
-  max: number;
-  min: number;
-  name: string;
-  params: AggParam[];
-  title: string;
-  defaults: unknown;
-  hideCustomLabel?: boolean;
-  mustBeFirst?: boolean;
-  aggSettings?: any;
-  disabled?: boolean;
-  tooltip?: ReactNode;
-}
+import { AggGroupNames, AggParam } from '../../data/public';
+import type { ISchemas, Schema } from '../../visualizations/public';
 
 export class Schemas implements ISchemas {
```

Comment:
discuss: I don't have a strong opinion here, but I'm not sure the `Schemas` class should stay in `vis_default_editor` since now..
Is there any reason to leave it here ?
Assuming interfaces were moved into visualizations, the class definition should be also moved there.

Reply:
good point, I've checked that and looks like you are right. `visualizations` is more suitable place for that. Initially I thought that we use it only in `vis_default_editor`

Reply:
I don't have a strong opinion either but I think that Schemas class is highly connected with the editor so I am not sure if it should be part of the visualizations plugin.

Reply:
@stratoula  I don't agree with you that it's more connected with `vis_default_editor`. From my point of view It's related to `visualizations` cause other plugins read this value though `public schemas` field  in `VisType` object (see _src/plugins/visualizations/public/vis_types/base_vis_type.ts_)

What I did it my last commit: 
- `Schemas` marked as private for `vizualizations` plugin and now we use it only from one place. In future we can probably remove it cause I don't see any reason to keep that class. We use it only for restructure initial `type.editorConfig.schemas` array. Perhaps we can create simple static functions to make this more obvious. 

@timroes please correct me if I'm wrong but I see our current design like on screen below (sorry for ugly picture).  The main idea here that our visualizations should not know about about the `visDefaultEditor` and use API-types from `visualizations` plugin. It allow us to simply change change the editor just by modifying schema in future

![image](https://user-images.githubusercontent.com/20072247/103643920-89144e80-4f66-11eb-89f2-91fc80e7e923.png)

Reply:
Ok Aliaksei, I see your point and the last change has made things a little bit better. I will approve it as it removes the dependencies between the plugins.

Reply:
I think `Schemas` were already a bit on the fence and still are unfortunately. While I think the sole purpose they are used today are the default editor and to build its UI (and thus I would say semantically they belong most into the default editor plugin), we returned them as a high level concept from the VisType as Alexej mentioned and thus they became a core concept of the vis registration itself (and should belong in visualizations).

So I would see two options:

* Change the API that Schemas will not be returned as a core concept of visualization registry, but be purely part of the editor, e.g. by passing it into a property of the default editor explciitly and always specifying the default editor explicitly (like `registerVisType({ ..., editor: () => <DefaultEditor schemas={schemas} />, ... })`) and then move then also in vis default editor and keep them there, or
* (the easier solution for now): don't change the APIs and keep them as they are, in which case I think they need to live in `Visualizations` for now, since they are a high level concern of VisType (and the registry).

Reply:
@timroes It's looks like a right things, but unfortunately not only the `vis_default_editor` uses schemas and it was a blocker why I didn't remove it at all in that PR.  At least one plugin `vis_type_xy` reads that value (see `src/plugins/vis_type_xy/public/editor/components/options/metrics_axes/index.tsx`) and to avoid any regression I decided to keep it in visualizations plugin. Semantically currently it's a best place for that.

From my point of view, we should:
-  `visualizations` plugin should have only `Schema` interface; It probably should be renamed to `EditorSchema`
- `Schemas` class and `ISchema` interface should be removed at all; 
- For `vis_default_editor` we should create simple functions to extract `metrics/buckets` from the `editorConfig.schemas ` in more obvious way. 
- Think why `vis_type_xy` uses schemas. Looks strange  

But it's out of that PR. If you agree with me or have your own plan to refactor that let's create a new item for that.